lib/rubocop/cop/chef/deprecation/legacy_notify_syntax.rb

```diff
@@ -56,7 +56,16 @@ def on_send(node)
           def autocorrect(node)
             lambda do |corrector|
               legacy_notify?(node) do |action, type, name, timing|
-                new_val = "notifies #{action.source}, '#{type.source}[#{name.str_type? ? name.value : name.source}]'"
+                name_val_string = case name.type
+                                  when :str
+                                    name.value
+                                  when :dstr
+                                    name.value.source # avoids double quotes around strings
+                                  else
+                                    "\#{#{name.source}}"
+                                  end
+                quote_type = name.str_type? ? "'" : '"'
```

Comment:
you might consider baking quote_type into the case statement directly. so you have one conditional that returns `"thing"`, one that returns `'thing'` and one that returns `"#{thing}"`

Reply:
Yep good point. The quote bit came first and then the case statement and I should have circled back to nuke that funky quote logic away.